src/allocator.c

```diff
@@ -25,3 +25,32 @@ allocator_config_init(allocator_config *allocator_cfg,
    uint64 log_extent_size         = 63 - __builtin_clzll(io_cfg->extent_size);
    allocator_cfg->extent_mask     = ~((1ULL << log_extent_size) - 1);
 }
+
+/*
+ * Return page number for the page at 'addr', in terms of page-size.
+ * This routine assume that input 'addr' is a valid page address.
+ */
+uint64
+allocator_page_number(allocator *al, uint64 page_addr)
```

Comment:
Concept of `page number` and `page offset` (see L49 below) are useful constructs to deal with logical pages, w/o worrying about page-size. I have used this (in future PRs) for improving diagnostics, which you will see in other reviews. 

I am not changing any existing interfaces that deal with `page address` and `extent address`, but simply providing another way to look at these addresses in terms of page-sized "things".

`page offset` is the index of the page in the holding extent; [0 .. 31].